crates/interpreter/src/instructions/memory.rs

```diff
@@ -68,3 +64,25 @@ pub fn mcopy<SPEC: Spec>(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     // copy memory in place
     interpreter.memory.copy(dest, src, len);
 }
+
+/// see https://eips.ethereum.org/EIPS/eip-4844
+pub fn blob_hash<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
+    check!(interpreter, SPEC::enabled(CANCUN));
+    // We add an instruction BLOBHASH (with opcode HASH_OPCODE_BYTE) which reads index from the top of the stack as big-endian uint256,
+    // and replaces it on the stack with tx.blob_versioned_hashes[index] if index < len(tx.blob_versioned_hashes),
+    // and otherwise with a zeroed bytes32 value. The opcode has a gas cost of HASH_OPCODE_GAS.
+    gas!(interpreter, gas::HASH_OPCODE_GAS);
+    pop!(interpreter, index);
+    let index = as_usize_or_fail!(interpreter, index, InstructionResult::InvalidOperandOOG);
```

Comment:
Should it fail or not?

Reply:
No i don't think it should, can i use `as_usize_saturated!()` here instead?